src/transformers/models/vocos/modeling_vocos.py

```
@@ -0,0 +1,302 @@
+    models.
+    """
+
+    config_class = VocosConfig
+    base_model_prefix = "vocos"
+    main_input_name = "audio_spectrogram"
+    supports_gradient_checkpointing = False
+
+    def _init_weights(self, module):
+        """Initialize the weights"""
+        if isinstance(module, nn.Linear):
+            std = getattr(self.config, "initializer_range", 0.02)
+            module.weight.data.normal_(mean=0.0, std=std)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        elif isinstance(module, nn.Conv1d):
+            nn.init.kaiming_normal_(module.weight)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, VocosAdaptiveLayerNorm):
+            if hasattr(module, "bias") and module.bias is not None:
+                module.bias.data.zero_()
+            if hasattr(module, "weight") and module.weight is not None:
+                module.weight.data.fill_(1.0)
+
+
+@auto_docstring(
+    custom_intro="""
+    Main Vocos model for neural vocoding for high-quality audio generation. This model can be paired with [`VocosProcessor`] to generate audio from either mel-spectrograms or EnCodec embeddings.
+    """
+)
+class VocosModel(VocosPreTrainedModel):
+    def __init__(self, config: VocosConfig):
+        super().__init__(config)
+        self.backbone = VocosBackbone(config)
+        self.head = VocosISTFTHead(config)
+        self._bandwidth_to_id = {bandwidth: id for id, bandwidth in enumerate(config.bandwidths)}
+        self.post_init()
+
+    @can_return_tuple
+    @auto_docstring
+    def forward(
+        self,
+        audio_spectrogram: Optional[torch.FloatTensor] = None,
+        input_features: Optional[torch.FloatTensor] = None,
```

and what if the user provides both? we silently use only audio_spectrogram. This happens because there is no reason to provide both, so such a use case has no reason to be integrated in the model forward signature. `input_feature` and `audio_spectrogram` are both `input_features` (without considering the renaming, let's keep it to input_features for now).